tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +95,364 @@ mvn package
+`AirlineIntegrationTest.java`
+This class contains integration tests for the application. It tests various functionalities like getting, creating, updating, and deleting airlines, as well as listing airlines by country and destination airport.
 
-Open the ProfileController file found in the controllers folder and navigate to the save method.  Let’s break this code down. First, in the save method signature we have a ProfileRequest object. This brings in the information passed to the controller in the body of the request and is used to get a profile variable from the ProfileRequest object's getProfile method. Note that we auto-generate a random UUID for the pid.
+### Repository
 
-```java
-public Profile getProfile() {
-  return new Profile(
-    UUID.randomUUID().toString(),
-    firstName,
-    lastName,
-    email,
-    password);
-}
-```
+`AirlineRepositoryImpl.java`
+This class implements the AirlineRepository interface. It interacts with the Couchbase database to perform CRUD operations on airline documents. It uses the Couchbase Java SDK to execute queries and operations.
 
-> *from Post method of models/ProfileRequest.java*
+### Model
 
-Rather than saving the password in the account object as plain text, we hash it with [Bcrypt](https://github.com/jeremyh/jBCrypt) in the setter of the Profile object:
+`Airline.java`
+This class represents the data model for an airline. It contains fields such as ID, type, name, IATA code, ICAO code, callsign, and country. The class uses annotations for validation.
 
-```java
-public String getPassword() {
-  return password;
-}
+### Controller
 
-public void setPassword(String password) {
-  this.password = BCrypt.hashpw(password, BCrypt.gensalt());
-}
-```
+`AirlineController.java`
+This class defines the RESTful API endpoints for managing airlines. It handles HTTP requests for creating, updating, deleting, and retrieving airlines. It also provides endpoints for listing airlines by various criteria.
 
-> *from models/Profile.java*
+### Service
 
-Our `profile` document is ready to be persisted to the database.  We create call to the `collection` using the local variable `profileCol` and then call the `insert` method  and passing it the UUID from the Profile's `getPid` method as the key. Once the document is inserted we then return the document saved and the result all as part of the same object back to the user.
+`AirlineServiceImpl.java`
+This class implements the AirlineService interface. It acts as an intermediary between the controller and repository, providing business logic for managing airlines.
 
-```java
-Profile profile = userProfile.getProfile();
-profileCol.insert(profile.getPid(), profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
-```
+## Route Specifications
 
-> *from controllers/ProfileController.java*
+### GET Route
```

Comment:
Is it fetching the route?

Reply:
Changed it to mapping to prevent any confusion